test/runner/e2e_update_rel_test.cpp

```diff
@@ -19,134 +19,372 @@ class UpdateRelTest : public DBTest {
+            "person" /* dstTable */, 2 /* srcID */, 5 /* dstID */, 100 /* lengthProp */)));
+        ASSERT_TRUE(conn->query(getInsertKnowsRelQuery("person" /* srcTable */,
+            "person" /* dstTable */, 3 /* srcID */, 8 /* dstID */, 700 /* lengthProp */)));
+        ASSERT_TRUE(conn->query(getUpdateRelQuery("person" /* srcTable */, "person" /* dstTable */,
+            "knows" /* relation */, 2 /* srcID */, 5 /* dstID */, "SET e.length = 20")));
+        ASSERT_TRUE(conn->query(getUpdateRelQuery("person" /* srcTable */, "person" /* relation */,
+            "knows" /* relation */, 3 /* srcID */, 8 /* dstID */, "SET e.length = 421")));
+        commitOrRollbackConnectionAndInitDBIfNecessary(isCommit, transactionTestType);
+        auto expectedResult = isCommit ? vector<string>{"20", "421"} : vector<string>{};
+        auto result =
+            conn->query("MATCH (p:person)-[e:knows]->(:person) WHERE p.ID > 0 RETURN e.length");
+        auto actualResult = TestHelper::convertResultToString(*result);
+        sortAndCheckTestResults(actualResult, expectedResult);
+    }
+
+    static constexpr uint64_t NUM_PERSON_KNOWS_PERSON_RELS = 2500;
+};
+
+TEST_F(UpdateRelTest, UpdateIntPropCommitNormalExecution) {
+    updateIntProp(true /* isCommit */, TransactionTestType::NORMAL_EXECUTION);
+}
+
+TEST_F(UpdateRelTest, UpdateIntPropCommitRecovery) {
+    updateIntProp(true /* isCommit */, TransactionTestType::RECOVERY);
+}
+
+TEST_F(UpdateRelTest, UpdateIntPropRollbackNormalExecution) {
+    updateIntProp(false /* isCommit */, TransactionTestType::NORMAL_EXECUTION);
+}
+
+TEST_F(UpdateRelTest, UpdateIntPropRollbacktRecovery) {
+    updateIntProp(false /* isCommit */, TransactionTestType::RECOVERY);
+}
+
+TEST_F(UpdateRelTest, UpdateStrPropCommitNormalExecution) {
+    updateStrProp(true /* isCommit */, TransactionTestType::NORMAL_EXECUTION);
+}
+
+TEST_F(UpdateRelTest, UpdateStrPropCommitRecovery) {
+    updateStrProp(true /* isCommit */, TransactionTestType::RECOVERY);
+}
+
+TEST_F(UpdateRelTest, UpdateStrPropRollbackNormalExecution) {
+    updateStrProp(false /* isCommit */, TransactionTestType::NORMAL_EXECUTION);
+}
+
+TEST_F(UpdateRelTest, UpdateStrPropRollbacktRecovery) {
+    updateStrProp(false /* isCommit */, TransactionTestType::RECOVERY);
+}
+
+TEST_F(UpdateRelTest, UpdateListPropCommitNormalExecution) {
+    updateListProp(true /* isCommit */, TransactionTestType::NORMAL_EXECUTION);
+}
+
+TEST_F(UpdateRelTest, UpdateListPropCommitRecovery) {
+    updateListProp(true /* isCommit */, TransactionTestType::RECOVERY);
+}
+
+TEST_F(UpdateRelTest, UpdateListPropRollbackNormalExecution) {
+    updateListProp(false /* isCommit */, TransactionTestType::NORMAL_EXECUTION);
+}
+
+TEST_F(UpdateRelTest, UpdateListPropRollbacktRecovery) {
+    updateListProp(false /* isCommit */, TransactionTestType::RECOVERY);
+}
+
+TEST_F(UpdateRelTest, UpdateEachElemOfSmallListCommitNormalExecution) {
```

Comment:
Since the test name is already long, we might type out Elem -> Element.

Reply:
changed `elem` -> `element`